### DISCUSSION TITLE
LGCP DIC and WAIC in experimental mode

### DISCUSSION THREAD

**Discussion by hannamcc** [inlabru-org/inlabru#179] [Q&A] [answered]
Hi all, 

I am working on a preferential sampling model (and a simpler LGCP for testing) in inlabru and have noticed the strange DIC/WAIC behavior mentioned in [this tutorial](https://inlabru-org.github.io/inlabru/articles/web/2d_lgcp_covars.html#non-spatial-evaluation-of-the-covariate-effect). (Running R 4.2.1, INLA 22.12.03, and inlabru 2.7.0 and I recover same values in classic & experimental mode as published in the tutorial)

It appears this issue is under investigation, but in the meantime, I am interested in learning more about what may be going on, what to look out for/words of caution, and any recommendations or workarounds for model comparison. Experimental mode has been performing better on simulated data for my application, and my sense is I should stick with experimental mode, but I see the tutorial went ahead with the classic mode DIC values for model selection. What was the reasoning behind this choice? 

Looking forward to any insights/discussion anyone may have!
Thanks, 
Hanna

**Comment by finnlindgren** [comment, marked as answer]
The tutorial was written long ago, so the part after the DIAC/WAIC classic/experimental mode comparison is still the original version; the comparison was added when we noticed the issue. I did some work to figure out what the correct values should be (and I believe the issue may have improved in recent inla versions; the vignette is rebuilt with every inlabru update, so typically shows results for recent INLA versions) but it's a tricky issue: Due to the way the LGCP likelihood construction works, it can only ever give relative comparisons, as there are normalisation constants that aren't included in the construction. Strangely, when manually working out what those normalisation constants would be, the resulting DIC/WAIC values were still not the same. I'm meeting Haavard in mid-January and hope to resolve the issue then.  From what I recall, at least one of DIC and WAIC may also not be fully well-defined for point process likelihoods (even assuming perfect calculations); that's also something to check.  Lately, we've worked on cross-validation approaches where one leaves out parts of space and uses prediction scores for the resulting Poisson mixture predictions. A vignette is almost done.

**Reply by hannamcc** [reply]
Good to know, thanks for the explanation! I'll look forward to the upcoming vignette. Since CPO does not rely on the estimating the likelihood in the same way as DIC, is it less impacted by the current behavior? Might it be a valid option for validation/comparison in this case , despite in general, its tendency to be unstable and sensitive to assumptions?

Thanks!

**Reply by finnlindgren** [reply]
No, on the contrary, CPO isn't defined _at all_ for point process likelihoods, as it involves "leave-one-out" cross validation, which has no meaning for the full point process likelihood. (For some specific discretisations of the likelihood it does have meaning, but that's just a special case, and does not hold for the more accurate likelihood construction that inlabru uses)  This should probably be flagged by inlabru; inla will "happily" go ahead and compute _something_ but it has no well-defined/known meaning.

**Reply by finnlindgren** [reply]
Come to think of it, that may be the issue with WAIC as well, as there is at least an _interpretation_ of that that relies on leave-one-out cross-validation.  Clearly there's more work to be done with this; apart from figuring out what's currently being computed, to decide what _should_ be computed.  Properly spatial cross-validation is something we should be able to do (we've started to do some research).

**Reply by hannamcc** [reply]
Ahh yes, that makes sense. Definitely an interesting problem in this case then, to think through how to compare preferential sampling models.

**Comment by finnlindgren** [comment]
I discussed this with Haavard today, and I'm leaning towards neither DIC
nor WAIC not being properly define for this (in their standard form). We
came up with a possible solution. We'll need to do some research and coding
to make that available.  What inla currently computes has a kind of
mathematical relation to something like what is needed, but it has no
theoretical basis, and there is no clear reason they should be useful. I'll
add a warning to inlabru about this.
